apps/project/serializers.py

```diff
@@ -674,8 +674,12 @@ def update(self, instance: Project, validated_data: dict[str, typing.Any]) -> Pr
             transaction.on_commit(lambda: process_project_task.delay(updated_project.pk))
 
         elif (
-            old_status_enum != Project.Status.READY_TO_PUBLISH
-            and updated_project.status_enum == Project.Status.READY_TO_PUBLISH
+            (
+                old_status_enum != Project.Status.READY_TO_PUBLISH
+                and updated_project.status_enum == Project.Status.READY_TO_PUBLISH
+            )
+            or (old_status_enum == Project.Status.PUBLISHED and updated_project.status_enum != Project.Status.PUBLISHED)
+            or (old_status_enum == Project.Status.PAUSED and updated_project.status_enum == Project.Status.PUBLISHED)
```

Comment:
Let's make this more generic.
The condition should be if the status has changed and either old or new status is any of "PUBLISHED", "PAUSED", "WITHDRAWN" or "FINISHED"